pkg/skaffold/kubernetes/manifest/images.go

```diff
@@ -62,9 +64,12 @@ func (rsi *ResourceSelectorImages) allowByGroupKind(gk apimachinery.GroupKind) b
 }
 
 func (rsi *ResourceSelectorImages) allowByNavpath(gk apimachinery.GroupKind, navpath string, k string) (string, bool) {
+	matchedConfigConnectorImage := false
+
 	for _, w := range ConfigConnectorResourceSelector {
-		if w.Matches(gk.Group, gk.Kind) {
-			return "", true
+		if k == imageField && w.Matches(gk.Group, gk.Kind) {
+			matchedConfigConnectorImage = true
+			break
```

Comment:
Clarifying question: is the reason we don't return here anymore to let the denylist stop it if the user configures that?

Reply:
Yes, but not only that. You cannot return here before also evaluating the allowlist below. The default `.*` will replace `image` fields, but if the user defines a selector with some field name, `image` won't be replaced (unless the user definition includes the `image` field as well). This is the behavior for other non-Config Connector resources as well. I wrote some tests to verify this behavior.